tests/unit/parallel/spmd_block.cpp

```diff
@@ -52,6 +56,42 @@ int main()
                 num_images, std::move(bulk_test), std::ref(c2));
 
     hpx::wait_all(join);
+*/
+
+    auto bulk_test =
+        [](hpx::parallel::v2::spmd_block block, boost::atomic<std::size_t> * cptr)
+        {
+            boost::atomic<std::size_t> & c = *cptr;
+
+            HPX_TEST_EQ(block.get_num_images(), num_images);
+            HPX_TEST_EQ(block.this_image() < num_images, true);
+
+            for(std::size_t i=0, test_count = num_images;
+                i<iterations;
+                i++, test_count+=num_images)
+            {
+                ++c;
+                block.sync_all();
+                HPX_TEST_EQ(c, test_count);
+                block.sync_all();
+            }
+        };
+
+    boost::atomic<std::size_t> * c1 = new boost::atomic<std::size_t>(0);
+    boost::atomic<std::size_t> * c2 = new boost::atomic<std::size_t>(0);
+
+    hpx::parallel::v2::define_spmd_block(
+        num_images, std::move(bulk_test), c1);
+
+    std::vector<hpx::future<void>> join =
+        hpx::parallel::v2::define_spmd_block(
+            par(task),
+                num_images, std::move(bulk_test), c2);
+
+    hpx::wait_all(join);
+
+    delete(c1);
+    delete(c2);
```

Comment:
Wouldn't it be much easier if you wrote:

    boost::atomic<std::size_t> c1(0);
    hpx::parallel::v2::define_spmd_block(  
         num_images, std::move(bulk_test), &c1);  

IIRC, there is no need to dynamically allocate the atomic variable.